crypto/src/utils.rs

```diff
@@ -0,0 +1,205 @@
+    file.write_all(file_content.as_bytes())?;
+
+    Ok(pk)
+}
+
+/// Reads `PublicKey` and `SecretKey` from encrypted file located by path and returns its.
+pub fn read_keys_from_file<P: AsRef<Path>>(
+    path: P,
+    pass_phrase: &[u8],
+) -> Result<(PublicKey, SecretKey), Error> {
+    let mut key_file = File::open(path)?;
+
+    if cfg!(unix) {
+        let file_info = key_file.metadata()?;
+        if (file_info.mode() & 0o600) != 0o600 {
+            return Err(Error::new(ErrorKind::Other, "Wrong file's mode"));
+        }
+    }
+
+    let mut file_content = vec![];
+    key_file.read_to_end(&mut file_content)?;
+    let keys: EncryptedKeys =
+        toml::from_slice(file_content.as_slice()).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    keys.decrypt(pass_phrase)
+}
+
+struct PublicKeyHex;
+
+impl Hex<PublicKey> for PublicKeyHex {
+    fn create_bytes(value: &PublicKey) -> Cow<[u8]> {
+        Cow::Borrowed(&*value.as_ref())
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Result<PublicKey, String> {
+        PublicKey::from_slice(bytes)
+            .ok_or_else(|| "Couldn't create PublicKey from slice".to_owned())
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct EncryptedKeys {
+    #[serde(with = "PublicKeyHex")]
+    public_key: PublicKey,
+    secret_key: ErasedPwBox,
+}
+
+impl EncryptedKeys {
+    fn encrypt(
+        public_key: PublicKey,
+        secret_key: &SecretKey,
+        pass_phrase: &[u8],
+    ) -> Result<EncryptedKeys, Error> {
+        let mut rng = thread_rng();
+        let mut eraser = Eraser::new();
+        let seed = &secret_key[..SEED_LENGTH];
+        eraser.add_suite::<Sodium>();
+        let pwbox = Sodium::build_box(&mut rng)
+            .seal(pass_phrase, &seed)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't create a pw box"))?;
+        let encrypted_key = eraser
+            .erase(&pwbox)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't convert a pw box"))?;
+
+        Ok(EncryptedKeys {
+            public_key,
+            secret_key: encrypted_key,
+        })
+    }
+
+    fn decrypt(self, pass_phrase: &[u8]) -> Result<(PublicKey, SecretKey), Error> {
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let restored = match eraser.restore(&self.secret_key) {
+            Ok(restored) => restored,
+            Err(_) => {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    "Couldn't restore a secret key",
+                ))
+            }
+        };
+        assert_eq!(restored.len(), SEED_LENGTH);
+        let seed_bytes = restored
+            .open(pass_phrase)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't open an encrypted key"))?
+            .to_vec();
+        let seed = Seed::from_slice(&seed_bytes[..])
+            .ok_or_else(|| Error::new(ErrorKind::Other, "Couldn't create seed from slice"))?;
```

Comment:
Minor nit: it seems that `Seed::from_slice` cannot fail if the slice length is correct (after all, [there is](https://docs.rs/exonum/0.9.4/exonum/crypto/struct.Seed.html#method.new) an infallible constructor from an array `[u8; SEED_LENGTH]`). But I'm not against being safe just in case 🙃